lib/ui/painting/flutter_rtree.h

```diff
@@ -0,0 +1,103 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_PAINTING_FLUTTER_RTREE_H_
+#define FLUTTER_LIB_UI_PAINTING_FLUTTER_RTREE_H_
+
+#include "third_party/skia/include/core/SkBBHFactory.h"
+#include "third_party/skia/include/core/SkTypes.h"
+
+namespace flutter {
+/**
+ * An R-Tree implementation based on
+ * https://chromium.googlesource.com/skia/+/master/src/core/SkRTree.h. In short,
+ * it is a balanced n-ary tree containing a hierarchy of bounding rectangles.
+ *
+ * It only supports bulk-loading, i.e. creation from a batch of bounding
+ * rectangles. This performs a bottom-up bulk load using the STR
+ * (sort-tile-recursive) algorithm.
+ *
+ * TODO: Experiment with other bulk-load algorithms (in particular the Hilbert
```

Comment:
Is this something we are going to investigate moving forward? If so, please file a bug for this and link it here.